core/src/main/java/io/questdb/griffin/engine/functions/groupby/CountLongGroupByFunctionFactory.java

```diff
@@ -55,11 +56,18 @@ public Function newInstance(
     ) throws SqlException {
         final Function arg = args.getQuick(0);
         if (arg.isConstant()) {
-            int val = arg.getInt(null);
-            // NULL expression would lead to zero matched rows, so it makes
-            // no sense to support it until we support count(expression).
-            if (val == Numbers.INT_NaN) {
-                throw SqlException.$(argPositions.getQuick(0), "NULL is not allowed");
+            if (arg instanceof LongConstant) {
```

Comment:
why do we have to use `instanceof` check? what can possibly go wrong here?

Reply:
That fixes cases like count(1L) that throw UnsupportedOperationException in master .
#2896 has a better fix